sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java

```diff
@@ -67,17 +68,20 @@ static ResultReturner forMethod(Class<?> extensionType, Method method)
                     method.getDeclaringClass().getName(),
                     method.getName())));
         }
-        else if (ResultIterable.class.isAssignableFrom(returnClass)) {
-            return new ResultIterableResultReturner(returnType);
+        else if (ResultIterable.class.equals(returnClass)) {
+            return new ResultIterableReturner(returnType);
         }
-        else if (Stream.class.isAssignableFrom(returnClass)) {
+        else if (Stream.class.equals(returnClass)) {
             return new StreamReturner(returnType);
         }
-        else if (Iterator.class.isAssignableFrom(returnClass)) {
-            return new IteratorResultReturner(returnType);
+        else if (ResultIterator.class.equals(returnClass)) {
+            return new ResultIteratorReturner(returnType);
+        }
+        else if (Iterator.class.equals(returnClass)) {
```

Comment:
I changed these from `isAssignableFrom()` to `equals()`, since the former would match any subtype of e.g. `Iterator` as the return type--which would cause a class cast exception.